cmake/defaults/CXXDefaults.cmake

```diff
@@ -25,8 +25,10 @@ include(CXXHelpers)
 include(Version)
 include(Options)
 
-# Require C++14
-set(CMAKE_CXX_STANDARD 14)
+# Default to C++14
+if (NOT DEFINED CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 14)
+endif()
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
```

Comment:
Probably worth giving these other CXX values the same treatment.

Reply:
I was hesitant to do so but I think you are right for `CMAKE_CXX_STANDARD_REQUIRED`: if a consuming project wants to build with C++20 but fallback to C++17 if 20 is not available, it should probably be allowed to do so. On the other hand I don't know if it makes sense to let `CMAKE_CXX_EXTENSIONS` be customizable considering USD doesn't use extensions. But maybe it can cause issues if some headers change depending on extensions, as they would be different in different translation units :thinking: . Let's leave it configurable as well.

Reply:
Done